src/App.tsx

```diff
@@ -0,0 +1,189 @@
+        cartPriceTotal: Number(searchParams.get('cartPriceTotal')) || 0,
+        cartAmountTotal: Number(searchParams.get('cartAmountTotal')) || 0,
+        minPrice: Number(searchParams.get('minPrice')) || appLib.getMinPrice(),
+        maxPrice: Number(searchParams.get('maxPrice')) || appLib.getMaxPrice(),
+        minStock: Number(searchParams.get('minStock')) || appLib.getMinStock(),
+        maxStock: Number(searchParams.get('maxStock')) || appLib.getMaxStock(),
+        categoryArr: searchParams.get('categoryArr') || appLib.getCategoryProd().map(obj => {
+            if (obj.view === true) {
+                return obj.category;
+            }
+        }).join(','),
+        brandArr: searchParams.get('brandArr') || appLib.getBrandProd().map(obj => {
+            if (obj.view === true) {
+                return obj.brand;
+            }
+        }).join(','),
+        sort: searchParams.get('sort') || 'default',
+        searchParam: searchParams.get('searchParam') || 'default',
+    });
+
+    const [products, setProducts] = useState(appLib.getProductsData());
+    const [prodDetails, setProdDetails] = useState(products[0]);
+    const [prodsInCart, setProdsInCart] = useState(appLib.addToCart(666));
+    const [cartPriceTotal, setCartPriceTotal] = useState(appLib.getCartPriceTotal());
+    const [cartAmountTotal, setCartAmountTotal] = useState(appLib.getCartAmountTotal());
+
+    useEffect(() => {
+        console.log('effect')
+        // @ts-ignore
+        setSearchParams(queryParams);
+        const catArr = queryParams.categoryArr.split(',');
+        const brArr = queryParams.brandArr.split(',');
+        applyFilters(
+            {
+                minPrice: Number(queryParams.minPrice),
+                maxPrice: Number(queryParams.maxPrice),
+                minStock: Number(queryParams.minStock),
+                maxStock: Number(queryParams.maxStock),
+                categoryArr: appLib.getCategoryProd().map(obj => {
+                    if (catArr.includes(obj.category)) {
+                        return { category: obj.category, view: true };
+                    }
+                    return { category: obj.category, view: false };
+                }),
+                brandArr: appLib.getBrandProd().map(obj => {
+                    if (brArr.includes(obj.brand)) {
+                        return { brand: obj.brand, view: true };
+                    }
+                    return { brand: obj.brand, view: false };
+                }),
+                sort: queryParams.sort,
+            });
+        applySearch(queryParams.searchParam);
+    }, []);
+
+
+    const btnHandler = (act: string, productId: number) => {
+        setProdsInCart(appLib.addToCart(productId));
+        setCartAmountTotal(appLib.getCartAmountTotal());
+        setCartPriceTotal(appLib.getCartPriceTotal());
+    };
+
+    const applyFilters = (viewParam: viewParam) => {
+        queryParams.minPrice = viewParam.minPrice;
```

Comment:
It will be much easier to read and to test code, if handler and filter functions were in a separate file.

And such usage of value from outer scope (I mean queryParams) is not a good practice. And it's a violation of functional programming paradigm.